assets/src/js/bindings/sources.js

```diff
@@ -24,54 +49,133 @@ const resolveImageAttribute = ( imageObj, attribute ) => {
 			return imageObj.alt_text || '';
 		case 'title':
 			return imageObj.title?.rendered || '';
+		case 'id':
+			return imageObj.id;
 		default:
 			return '';
 	}
 };
 
+/**
+ * Handle object-type field values (like complex field objects).
+ *
+ * @param {Object} fieldValue The field value object.
+ * @param {string} attribute The attribute being processed.
+ * @returns {string} The resolved value.
+ */
+const handleObjectFieldValue = ( fieldValue, attribute ) => {
+	// Check if the field value has the exact attribute property
+	if ( fieldValue.hasOwnProperty( attribute ) && fieldValue[ attribute ] ) {
+		return fieldValue[ attribute ];
+	}
+
+	// Special fallback: if we're looking for 'content' and no content property exists (or is falsy),
+	// but there's a 'url' property, use that instead
+	if ( attribute === 'content' && fieldValue.url ) {
+		return fieldValue.url;
+	}
```

Comment:
Is this for images?

Reply:
Image doesn't have 'content' as a bindable attribute, lets change that.